projects/transformers/run.py

```diff
@@ -576,27 +614,31 @@ def run_finetuning_multiple_tasks(
         )
 
     base_training_args = deepcopy(training_args)
+    base_model_args = deepcopy(model_args)
     for task_name in data_args.task_names:
         data_args.task_name = task_name
         training_args = deepcopy(base_training_args)
+        model_args = deepcopy(base_model_args)
         # For each task, save to a subfolder within run's root folder
         training_args.run_name = f"{base_training_args.run_name}_{task_name}"
         training_args.output_dir = os.path.join(
             base_training_args.output_dir, task_name
         )
 
+        model_arg_keys = ["trainer_class", "task_hyperparams_proxy"]
```

Comment:
Is that the case we have different trainer_class for each finetuning task? If not, and task_hyperparams_proxy is the only argument that changes, couldn't that also be part of training_args so we don't have to include model_args in the hp loop as well?

Reply:
The same trainer class is used for all tasks except mnli, which requires a class that uses the multi_eval_sets mixin. Model args is also needed because ("hp_" in hp_key) is checking to update things like hp_num_trials and hp_compute_objective. I admit this is a little messy, but I'm prioritizing gathering results atm, so I'm gonna roll with it.